packages/cadl-ranch-specs/http/client/initialization/basic/main.tsp

```diff
@@ -0,0 +1,31 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+/**
+ * Describe client with `@clientInitialization`
+ */
+@supportedBy("dpg")
+@scenarioService("/client/initialization/basic")
+@scenario
+@scenarioDoc("""
+  Testing that "SubClient" be initialized with "name" client property,
+  and that the "action" API under "SubClient" does not take "name" parameter in method signature.
+  
+  Use "subclient1" as "name" parameter in test.
+  """)
+namespace Client.Initialization.Basic;
+
+model SubClientOptions {
+  name: string;
+}
+
+@clientInitialization(SubClientOptions)
+interface SubClient {
+  @post
+  @route("/sub-client/{name}:action")
+  action(@path name: string): void;
```

Comment:
I'm not sure, maybe add another parameter that will be taken in method signature, as a contrast?

Reply:
Sure, I will first update to our local test first, as this one likely going to take time.
https://github.com/microsoft/typespec/pull/5027